dot/network/notifications.go

```diff
@@ -419,6 +419,7 @@ func (s *Service) broadcastExcluding(info *notificationsProtocol, excluding peer
 			continue
 		}
 
+		info.outboundHandshakeMutexes.Store(peer, new(sync.Mutex))
```

Comment:
don't think we need to add this here, as it's already stored in `sendHandshake`

Reply:
I don't see any code storing mutexes in sendhandshake.

In fact, I added this line because sendHandshake was not able to load these mutexes.

https://github.com/ChainSafe/gossamer/blob/e4bc3757c2011a387d34baba64c8a5bb9aee1a81/dot/network/notifications.go#L321

Reply:
sorry not here I meant in there's a function to store them in the ConnManager in `Start()`:
```

	// this handles all new connections (incoming and outgoing)
	// it creates a per-protocol mutex for sending outbound handshakes to the peer
	s.host.cm.connectHandler = func(peerID peer.ID) {
		for _, prtl := range s.notificationsProtocols {
			prtl.outboundHandshakeMutexes.Store(peerID, new(sync.Mutex))
		}
	}
```

so the mutex should already be stored, as it's created on peer connection

Reply:
were you seeing `errMissingHandshakeMutex` being returned?

Reply:
> were you seeing errMissingHandshakeMutex being returned?

Sorry, missed this comment. Yes, I have seen this being returned a lot of times!